earthfile2llb/interpreter.go

```diff
@@ -952,11 +952,11 @@ func (i *Interpreter) handleBuild(ctx context.Context, cmd spec.Command, async b
 	if i.local && !(isSafeAsyncBuildArgsKVStyle(opts.BuildArgs) && isSafeAsyncBuildArgs(args[1:])) {
```

Comment:
I wonder if this kind of check is still necessary now. Perhaps it's only necessary when the shell-out-anywhere feature flag is turned off, right?

Reply:
That's correct -- when we enable `--shell-out-anywhere` for all and make it impossible for the other case, these checks can be removed.

and in the case of the `i.local`, we might even be able to pass a callback into the `c.varCollection.Expand` in order to support shelling-out.